components/image/slider/src/index.js

```diff
@@ -1,45 +1,73 @@
+// import IconCamera from '@schibstedspain/sui-svgiconset/lib/Camera'
 
-const ImageSlider = (props) => {
-  const slides = getSlides(props.images, props.linkFactory)
+const NO_OP = () => {}
 
-  return (
-    <div onClick={props.handleClick} className='sui-ImageSlider'>
-      { hasMoreThanOneImage(props.images)
-        ? (<ReactSlidy {...props.sliderOptions} dynamicContent={props.dynamicContent}>{ slides }</ReactSlidy>)
-        : slides
-      }
-    </div>
-  )
-}
+class ImageSlider extends Component {
+  constructor (props) {
+    super(props)
+    const {sliderOptions} = props
+    // In order to accept a custom doAfterSlide callback, and to avoid altering props.sliderOptions
+    // sliderOptions is deeply cloned.
+    this._sliderOptions = cloneDeep(sliderOptions)
+    this._sliderOptions.doAfterSlide = (currentSlide) => {
+      this.setState(currentSlide)
+      sliderOptions.doAfterSlide && sliderOptions.doAfterSlide(currentSlide)
+    }
+  }
+
+  state = {currentSlide: 0}
+
+  render () {
+    const {images, linkFactory, handleClick, dynamicContent, enableCounter} = this.props
+    const slides = this._getSlides(images, linkFactory)
+    return (
+      (slides.length > 0) &&
+      <div onClick={handleClick} className='sui-ImageSlider'>
+        { (slides.length > 1)
+          ? (<ReactSlidy {...this._sliderOptions} dynamicContent={dynamicContent}>{ slides }</ReactSlidy>)
+          : slides
+        }
+        {enableCounter && this._buildCounter(slides.length)}
+      </div>
+    )
+  }
 
-/**
- * Says if there are more than one image in the given list.
- * Implemented for semantic purposes.
- * @param {Array} images List given by props.images.
- * @return {boolean}
- */
-const hasMoreThanOneImage = (images) => (images && images.length > 1)
+  _buildCounter (totalImages) {
+    const classNames = cx('sui-ImageSlider-counter', `sui-ImageSlider-counter--${this.props.counterPosition}`)
+    const Icon = this.props.counterIcon
+    return <div className={classNames}>
+      <Icon svgClass='sui-ImageSlider-counterIcon' />
+      <span className='sui-ImageSlider-counterText'>{this.state.currentSlide + 1}/{totalImages}</span>
+    </div>
+  }
 
-/**
- * @param {Array} images List given by props.images.
- * @return {Array} List of img elements.
- */
-const getSlides = (images, linkFactory) => {
-  if (images && images.length) {
-    return images.map((image, index) => {
-      const key = image.key ? image.key + index : index
-      const img = <img className='sui-ImageSlider-image' key={key} src={image.src} alt={image.alt} />
-      const target = image.target ? image.target : '_blank'
-      return image.link ? linkFactory({ href: image.link, target: target, className: '', children: img, key: key }) : img
-    })
-  } else {
-    return []
+  /**
+   * @param {Array} images List given by props.images.
+   * @return {Array} List of img elements.
+   */
+  _getSlides (images, linkFactory) {
+    if (images && images.length) {
```

Comment:
If you perform this check in the line 28 of this component, you can skip this `if else`.

Reply:
And then take advantage of arrow function syntax ;)

Reply:
Actually I prefer this way because it makes this function returns always an array, which is a simply and clear contract that simplifies the render() code and also avoids writing another if/else or ternary operator whenever _getSlides was called (even when it is only called once now by now). Also, the if/else would not be skipped but written in a different manner in another place, with a ternary operator. I see no profit here.